openapi-core/src/main/java/org/openapi4j/core/validation/ValidationResults.java

```diff
@@ -20,10 +21,8 @@
 
   // The validation items
   private final List<ValidationItem> items = new ArrayList<>();
-  // The data breadcrumb
-  private final Deque<String> dataCrumbs = new ArrayDeque<>();
   // The schema breadcrumb
-  private final Deque<String> schemaCrumbs = new ArrayDeque<>();
+  private final Deque<CrumbInfo> crumbs = new ArrayDeque<>();
```

Comment:
Without concrete and useful usage there's no need to open it.
It sounds like a very a bad idea in all cases anyway to work with this queue directly.

Reply:
The use case is;
a. navigate to the referenced data without performing string parsing.
b. output the crumb information in a customised format.

Imagine a tool which validates the data then allows jumping to the point in the data where the validation error/warning/info is - or maybe just print out more information about the nodes value. It would be much easier if they had a structure to work with rather than a string. I am performing conversion using the crumb information so have similar needs - being able to pinpoint the node in the data from the crumb. (although I am happy to work with the string).

Perhaps it could return a new Array rather than a reference to the underyling queue.  Or return an unmodifiable/immutable wrapper for it: https://docs.oracle.com/javase/6/docs/api/java/util/Collections.html#unmodifiableCollection(java.util.Collection)

Reply:
note this is a low priority suggestion - needing to parse the string is not much of an issue.

Reply:
Well, we can open it if it's readonly. I did not want to get impossible issues to understand because of messing with the queue.

Reply:
It could be nice but I guess this won't happen because of global impact on performance and memory for limited/edgy usages